zp-relayer/validateTx.ts

```diff
@@ -145,18 +143,19 @@ export function checkLimits(limits: Limits, amount: BN) {
   return null
 }
 
-async function checkDepositEnoughBalance(address: string, requiredTokenAmount: BN) {
+async function checkDepositEnoughBalance(token: Contract, address: string, requiredTokenAmount: BN) {
   if (requiredTokenAmount.lte(toBN(0))) {
     throw new TxValidationError('Requested balance check for token amount <= 0')
   }
 
-  return checkBalance(address, requiredTokenAmount.toString(10))
+  return checkBalance(token, address, requiredTokenAmount.toString(10))
 }
 
 async function getRecoveredAddress(
   txType: TxType,
   proofNullifier: string,
   txData: TxData,
+  tokenContract: Contract,
```

Comment:
Looks like unused argument

Reply:
Token contract is used to fetch `nonce` below in this function